lib/bundler.rb

```diff
@@ -352,9 +352,9 @@ def load_gemspec_uncached(file, validate = false)
         spec
       end
     rescue Gem::InvalidSpecificationException => e
-      UI::Shell.new.warn "The gemspec at #{file} is not valid. " \
-        "The validation error was '#{e.message}'"
-      nil
+      UI::Shell.new.warn "The gemspec at #{file} is not valid. Please fix this gemspec.\n" \
+        "The validation error was '#{e.message}'\n"
+      raise Gem::InvalidSpecificationException
```

Comment:
Well, the original implementation would return `nil` and would hence iterate through all local gemspecs and presumably report all the validation errors with all those local gemspecs presumably. By changing to raise an error here instead of `nil`, we lose that ability to report on all invalid local gemspecs at once (how big of a user scenario is this?) but prevent fetching or further post-gemspec loading actions from occurring. 

Though, as an alternative, raising a new `Gem::InvalidSpecificationException` in this location with the full error message (what is currently in warn), is functionally equivalent in terms of what the end user sees I believe.